diff -u rcssserver-19.0.0/src/player.cpp rcssserver-19.0.0/src/player.cpp
--- rcssserver-19.0.0/src/player.cpp	2024-03-24 19:29:40
+++ rcssserver-19.0.0/src/player.cpp	2026-03-04 03:48:22
@@ -799,6 +799,13 @@
 
             goalieCatch( dir );
         }
+        else if ( ! std::strncmp( buf, "(drop)", 6 ) )
+        {
+            if ( this == M_stadium.ballCatcher() )
+            {
+                M_stadium.clearBallCatcher();
+            }
+        }
         else if ( ! std::strncmp( buf, "(move ", 6 ) )
         {
             double x, y;
@@ -1249,8 +1256,6 @@
         M_angle_body = normalize_angle( angleBodyCommitted()
                                         + ( 1.0 + drand( -M_randp, M_randp ) ) * omega );
     }
-
-    M_stamina = std::max( 0.0, stamina() - consumed_stamina );
 }
 
 // void
@@ -1535,11 +1540,11 @@
     //pfr: we should only be able to catch in PlayOn mode
     //tom: actually the goalie can catch the ball in any playmode, but
     //infringements should be awarded.  Maybe later.
-    if ( ! this->isGoalie()
-         || M_goalie_catch_ban > 0
+    if ( this->isGoalie() &&
+         ( M_goalie_catch_ban > 0
          || ( M_stadium.playmode() != PM_PlayOn
               && ! Referee::isPenaltyShootOut( M_stadium.playmode() ) )
-         )
+         ))
     {
         M_state |= CATCH_FAULT;
         return;
@@ -1674,6 +1679,7 @@
         M_vel = PVector();
 
         M_stadium.ballCaught( *this );
+        M_stadium.setLastCatcher( this );
     }
     else
     {
@@ -2733,6 +2739,25 @@
 bool
 Player::ballKickable() const
 {
+    // ensure that the player is in front of the ball first (20 degree margin)
+
+    // Facing = Player's facing direction unit vector | F = ( cos(angle_body), sin(angle_body) )
+    // Difference = Ball - Player | D = B - P
+    PVector facing = PVector::fromPolar( 1.0, Deg2Rad( M_angle_body ) );
+    PVector difference = M_stadium.ball().pos() - pos();
+    
+    // (F . D) = |F| * |D| * cos(angle_diff)
+    // Since |F| = 1, we can simplify this to:
+    // cos(angle_diff) = (F . D) / |D|
+    double dot_product = (facing.x * difference.x) + (facing.y * difference.y);
+    double angle_diff = std::acos( dot_product / difference.r() );
+    angle_diff = Rad2Deg( angle_diff );
+
+    if ( std::fabs( angle_diff ) > 20.0 )
+    {
+        return false;
+    }
+
     return pos().distance2( M_stadium.ball().pos() ) <= std::pow( kickableArea(), 2 );
 }
 
diff -u rcssserver-19.0.0/src/rcsoccersim.in rcssserver-19.0.0/src/rcsoccersim.in
--- rcssserver-19.0.0/src/rcsoccersim.in	2024-02-18 23:29:46
+++ rcssserver-19.0.0/src/rcsoccersim.in	2026-03-01 01:51:10
@@ -23,6 +23,7 @@
      echo "  -help                 print this message"
      echo "  -rcssmonitor          use rcssmonitor"
      echo "  -monitor MONITOR      specify the monitor command"
+     echo "  -port PORT            specify the port number for the server (default: 6000)"
     ) 1>&2
 }
 
@@ -37,6 +38,7 @@
     exit 1
 fi
 
+MON=`which rcssmonitor 2> /dev/null`
 if test $# -eq 0; then
     if test "$RCSSMONITOR"; then
         MON="$RCSSMONITOR"
@@ -84,6 +86,18 @@
                     echo ""
                     exit 1
                 fi
+                shift 1
+                ;;
+            -port)
+                PORT="${2}"
+                COACH_PORT=`expr $PORT + 1`
+                OLCOACH_PORT=`expr $PORT + 2`
+
+                SERV="$SERV --server::port=$PORT"
+                SERV="$SERV --server::coach_port=$COACH_PORT"
+                SERV="$SERV --server::olcoach_port=$OLCOACH_PORT"
+
+                MON="$MON --server-port $PORT"
                 shift 1
                 ;;
 
diff -u rcssserver-19.0.0/src/referee.cpp rcssserver-19.0.0/src/referee.cpp
--- rcssserver-19.0.0/src/referee.cpp	2024-02-18 23:29:46
+++ rcssserver-19.0.0/src/referee.cpp	2025-12-18 17:22:21
@@ -2477,7 +2477,7 @@
 void
 CatchRef::ballCaught( const Player & catcher )
 {
-    if ( isPenaltyShootOut( M_stadium.playmode() ) )
+    if ( catcher.isGoalie() == false  || isPenaltyShootOut( M_stadium.playmode() ) )
     {
         return;
     }
@@ -2533,7 +2533,7 @@
 void
 CatchRef::ballPunched( const Player & catcher )
 {
-    if ( isPenaltyShootOut( M_stadium.playmode() ) )
+    if ( catcher.isGoalie() == false || isPenaltyShootOut( M_stadium.playmode() ) )
     {
         return;
     }
@@ -2666,6 +2666,7 @@
 
 
     if ( M_stadium.ballCatcher()
+         && M_stadium.ballCatcher()->isGoalie()
          && pm != PM_AfterGoal_Left
          && pm != PM_AfterGoal_Right
          && pm != PM_TimeOver
diff -u rcssserver-19.0.0/src/referee.h rcssserver-19.0.0/src/referee.h
--- rcssserver-19.0.0/src/referee.h	2024-02-18 23:29:46
+++ rcssserver-19.0.0/src/referee.h	2025-12-18 17:22:11
@@ -785,9 +785,10 @@
 
     void playModeChange( PlayMode pm ) override;
 
+    void callFoul( const Player & tackler );
+
 private:
 
-    void callFoul( const Player & tackler );
     void callYellowCard( const Player & tackler );
     void callRedCard( const Player & tackler );
 
diff -u rcssserver-19.0.0/src/serverparam.cpp rcssserver-19.0.0/src/serverparam.cpp
--- rcssserver-19.0.0/src/serverparam.cpp	2024-03-24 19:29:40
+++ rcssserver-19.0.0/src/serverparam.cpp	2026-03-01 01:15:33
@@ -129,18 +129,18 @@
 const int ServerParam::HALF_TIME = 300;
 const int ServerParam::DROP_TIME = 100;
 
-const double ServerParam::PITCH_LENGTH = 105.0;
-const double ServerParam::PITCH_WIDTH = 68.0;
-const double ServerParam::PITCH_MARGIN = 5.0;
-const double ServerParam::CENTER_CIRCLE_R = 9.15;
-const double ServerParam::PENALTY_AREA_LENGTH = 16.5;
-const double ServerParam::PENALTY_AREA_WIDTH = 40.32;
-const double ServerParam::GOAL_AREA_LENGTH = 5.5;
-const double ServerParam::GOAL_AREA_WIDTH = 18.32;
-const double ServerParam::GOAL_WIDTH = 14.02;
-const double ServerParam::GOAL_DEPTH = 2.44;
+const double ServerParam::PITCH_LENGTH = 90.0;
+const double ServerParam::PITCH_WIDTH = 60.0;
+const double ServerParam::PITCH_MARGIN = 3.0;
+const double ServerParam::CENTER_CIRCLE_R = 5.0;
+const double ServerParam::PENALTY_AREA_LENGTH = 10.0;
+const double ServerParam::PENALTY_AREA_WIDTH = 20.0;
+const double ServerParam::GOAL_AREA_LENGTH = 3.6;
+const double ServerParam::GOAL_AREA_WIDTH = 12.5;
+const double ServerParam::GOAL_WIDTH = 10.0;
+const double ServerParam::GOAL_DEPTH = 1.8;
 const double ServerParam::PENALTY_SPOT_DIST = 11.0;
-const double ServerParam::CORNER_ARC_R = 1.0;
+const double ServerParam::CORNER_ARC_R = 0.0;
 const double ServerParam::KICK_OFF_CLEAR_DISTANCE = CENTER_CIRCLE_R;
 
 const double ServerParam::CORNER_KICK_MARGIN = 1.0;
@@ -148,18 +148,18 @@
 const double ServerParam::KEEPAWAY_LENGTH = 20.0;
 const double ServerParam::KEEPAWAY_WIDTH = 20.0;
 
-const double ServerParam::BALL_SIZE = 0.085;
+const double ServerParam::BALL_SIZE = 0.215;
 const double ServerParam::BALL_DECAY = 0.94;
-const double ServerParam::BALL_RAND = 0.05;
+const double ServerParam::BALL_RAND = 0.01;
 const double ServerParam::BALL_WEIGHT = 0.2;
 const double ServerParam::BALL_T_VEL = 0.001;
 const double ServerParam::BALL_SPEED_MAX = 3.0; // [12.0.0] 2.7 -> 3.0;
 const double ServerParam::BALL_ACCEL_MAX = 2.7;
 
-const double ServerParam::PLAYER_SIZE = 0.3;
+const double ServerParam::PLAYER_SIZE = 0.9;
 const double ServerParam::PLAYER_WIDGET_SIZE = 1.0;
 const double ServerParam::PLAYER_DECAY = 0.4;
-const double ServerParam::PLAYER_RAND = 0.1;
+const double ServerParam::PLAYER_RAND = 0.05;
 const double ServerParam::PLAYER_WEIGHT = 60.0;
 const double ServerParam::PLAYER_SPEED_MAX = 1.05; // [13.0.0] 1.2 -> 1.05
 // th 6.3.00
@@ -178,7 +178,7 @@
 const double ServerParam::EFFORT_INC_THR = 0.6;
 const double ServerParam::EFFORT_INC = 0.01;
 
-const double ServerParam::KICK_RAND = 0.1; // [12.0.0] 0.0 -> 0.1
+const double ServerParam::KICK_RAND = 0.01; // [12.0.0] 0.0 -> 0.1
 const double ServerParam::PRAND_FACTOR_L = 1.0;
 const double ServerParam::PRAND_FACTOR_R = 1.0;
 const double ServerParam::KICK_RAND_FACTOR_L = 1.0;
@@ -245,11 +245,11 @@
 const int ServerParam::GAME_LOGGING = true;
 const int ServerParam::GAME_LOG_VERSION = DEFAULT_REC_VERSION;
 #ifdef RCSS_WIN
-const std::string ServerParam::TEXT_LOG_DIR = ".\\";
-const std::string ServerParam::GAME_LOG_DIR = ".\\";
+const std::string ServerParam::TEXT_LOG_DIR = ".\\text_logs\\";
+const std::string ServerParam::GAME_LOG_DIR = ".\\game_logs\\";
 #else
-const std::string ServerParam::TEXT_LOG_DIR = "./";
-const std::string ServerParam::GAME_LOG_DIR = "./";
+const std::string ServerParam::TEXT_LOG_DIR = "./text_logs/";
+const std::string ServerParam::GAME_LOG_DIR = "./game_logs/";
 #endif
 const std::string ServerParam::TEXT_LOG_FIXED_NAME = "rcssserver";
 const std::string ServerParam::GAME_LOG_FIXED_NAME = "rcssserver";
@@ -352,8 +352,8 @@
 const double ServerParam::MAX_DASH_ANGLE = +180.0;
 const double ServerParam::MIN_DASH_ANGLE = -180.0;
 const double ServerParam::DASH_ANGLE_STEP = 1.0; // [15.6.0] 45.0 -> 1.0 [14.0.0] 90.0 -> 45.0
-const double ServerParam::SIDE_DASH_RATE = 0.4; // [14.0.0] 0.25 -> 0.4
-const double ServerParam::BACK_DASH_RATE = 0.7; // [14.0.0] 0.5 -> 0.6 [17.0.0] 0.6 -> 0.7
+const double ServerParam::SIDE_DASH_RATE = 1.0; // [14.0.0] 0.25 -> 0.4
+const double ServerParam::BACK_DASH_RATE = 1.0; // [14.0.0] 0.5 -> 0.6 [17.0.0] 0.6 -> 0.7
 const double ServerParam::MAX_DASH_POWER = +100.0;
 const double ServerParam::MIN_DASH_POWER = 0.0;
 
@@ -1307,7 +1307,7 @@
     M_catch_ban_cycle = GOALIE_CATCH_BAN_CYCLE;
     M_slow_down_factor = 1;
 
-    M_use_offside = true;
+    M_use_offside = false;
     M_forbid_kick_off_offside = true;
     M_offside_kick_margin = OFFSIDE_KICK_MARGIN;
 
@@ -1335,7 +1335,7 @@
     M_verbose = false;
 
     M_coach_mode = false;
-    M_coach_with_referee_mode = false;
+    M_coach_with_referee_mode = true;
     M_old_coach_hear = false;
 
     M_synch_mode = false;
diff -u rcssserver-19.0.0/src/stadium.cpp rcssserver-19.0.0/src/stadium.cpp
--- rcssserver-19.0.0/src/stadium.cpp	2024-03-24 19:29:40
+++ rcssserver-19.0.0/src/stadium.cpp	2025-12-19 16:20:17
@@ -74,6 +74,8 @@
       M_time( 0 ),
       M_stoppage_time( 0 ),
       M_ball_catcher( static_cast< const Player * >( 0 ) ),
+      M_last_catcher( static_cast< Player * >( 0 ) ),
+      M_ball_caught_loc( nullptr ),
       M_kick_off_side( LEFT ),
       M_last_playon_start( 0 ),
       M_game_over_wait( 0 ),
@@ -865,14 +867,6 @@
     //
     // update stamina etc
     //
-    for ( PlayerCont::reference p : M_players )
-    {
-        if ( ! p->isEnabled() ) continue;
-
-        p->updateStamina();
-        p->updateCapacity();
-    }
-
     if ( stoppageTime() == 0
          && time() > 0
          && ServerParam::instance().nrNormalHalfs() > 0
@@ -933,6 +927,19 @@
                                            M_ball_catcher->angleBodyCommitted() );
         M_ball->moveTo( M_ball_catcher->pos() + rpos );
     }
+
+    if ( M_last_catcher )
+    {
+        PVector ball_pos = M_ball->pos();
+        if ( ball_pos.distance( M_last_catcher->pos() ) > ServerParam::instance().catchAreaLength() )
+        {
+            clearLastCatcher();
+        }
+        else if ( ball_pos.distance( *M_ball_caught_loc ) > ServerParam::CENTER_CIRCLE_R * 2.0 )
+        {
+            callFoul( static_cast<Side>(-M_last_catcher->side()), ball_pos);
+        }
+    }
 }
 
 
@@ -1943,9 +1950,10 @@
               {
                   ref->ballCaught( catcher );
               } );
-
+    
     if ( playmode() == PM_FreeKick_Left
-         || playmode() == PM_FreeKick_Right )
+         || playmode() == PM_FreeKick_Right 
+         || catcher.isGoalie() == false)
     {
         M_ball_catcher = &catcher;
     }
diff -u rcssserver-19.0.0/src/stadium.h rcssserver-19.0.0/src/stadium.h
--- rcssserver-19.0.0/src/stadium.h	2024-02-18 23:29:46
+++ rcssserver-19.0.0/src/stadium.h	2025-12-18 23:09:12
@@ -120,6 +120,8 @@
 
 
     const Player * M_ball_catcher; /* goalie who has caught ball */
+    Player * M_last_catcher; /* last player who caught ball */
+    PVector * M_ball_caught_loc; /* location where the ball was caught */
 
     Side M_kick_off_side;
 
@@ -267,6 +269,26 @@
     void clearBallCatcher()
       {
           M_ball_catcher = static_cast< Player * >( 0 );
+      }
+
+    void setLastCatcher( Player * player )
+      {
+          if ( M_last_catcher == player ) return;
+          M_last_catcher = player;
+          PVector ball_pos = M_ball->pos();
+          M_ball_caught_loc = new PVector( ball_pos.x, ball_pos.y );
+      }
+
+    void clearLastCatcher()
+      {
+          M_last_catcher = static_cast< Player * >( 0 );
+          delete M_ball_caught_loc;
+          M_ball_caught_loc = nullptr;
+      }
+
+    PVector * ballCaughtLocation() const
+      {
+          return M_ball_caught_loc;
       }
 
     Side kickOffSide() const
